concourse/steps/update_component_deps.py

```diff
@@ -205,18 +205,44 @@ def determine_reference_versions(
         raise NotImplementedError
 
 
+def create_ctx_repo_mapping(
```

Comment:
might inline to modelclass; alternativ: make modelclass callable so it can directly be passed as value, like so:

```
@dataclass.dataclass
class OcmRepositoryLookup:
  lookups: list[OcmRepositoryPrefixMapping]

 def __call__(self, component_id):
   for lookup in self.lookups:
     if lookup.matches(component_id):
       return lookup.ocm_repo
```

(Almost pseudo-code)

That way, instances of this type work as any other callable (thus no conversion required)